### PR TITLE
Use `await` when calling functions on user-event

### LINKED ISSUE #11749: Use `await` when calling functions on `user-event`
We should always be `await`ing the result of a call to a `user-event` function. This will allow us to remove a lot of the `waitFor`s that we currently have in our tests because the component will have been updated when the promise from the `user-event` function returns.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #11749 

Also includes some additional test cleanup around calls to `waitFor`.

### CODE REVIEW

**Comment by bobsilverberg** [comment]
@diox It seems that, if we make all of the test changes that introduce `await`s for `user-event` function calls, but *do not* update the `user-event` library to `v14`, everything is fine. I've tried some other combinations, and it definitely seems that it's the upgrade to `v14` that is introducing this intermittent failure (as well as another, more rare one). We will want to update to `v14` at some point, but it's not terribly important to update to it right now, so I'm thinking we can land this commit, that cleans up all the tests, and then deal with those failing tests in a separate effort that is just about updating the library.

WDYT?

**Review by diox — APPROVED** [review]

File: tests/unit/amo/components/TestScreenShots.js [hunk 1]
```diff
@@ -96,19 +96,25 @@ describe(__filename, () => {
   Commenting this out for now.
   See https://github.com/mozilla/addons-frontend/issues/11482.
 
-  it('scrolls to the active item on close', () => {
+  it('scrolls to the active item on close', async () => {
```
[the remark this answers is not in the document]
Comment (diox) [reply tests/unit/amo/components/TestScreenShots.js:99]: (Doesn't seem to make a difference)

File: tests/unit/amo/components/TestInstallButtonWrapper.js [hunk 2]
```diff
@@ -1227,11 +1225,9 @@ describe(__filename, () => {
...
 
-        userEvent.click(screen.getByRole('link', { name: 'Enable' }));
+        await userEvent.click(screen.getByRole('link', { name: 'Enable' }));
 
-        await waitFor(() => {
-          expect(_addonManager.enable).toHaveBeenCalledWith(addon.guid);
-        });
+        expect(_addonManager.enable).toHaveBeenCalledWith(addon.guid);
 
         expect(dispatch).not.toHaveBeenCalledWith(
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply tests/unit/amo/components/TestInstallButtonWrapper.js:1232]: I'm not sure this is the case, or maybe I'm misunderstanding your point. The `expect(_addonManager.enable).toHaveBeenCalledWith(addon.guid);` ensures that `enable` was called, so there's no need for the `waitFor` around it, but maybe you're not talking about the removal of the `waitFor` and rather the fact that the `dispatch` that we want to prove doesn't get called might not have been called yet at this point?

File: tests/unit/amo/components/TestScreenShots.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply tests/unit/amo/components/TestScreenShots.js:99]: Yes, as you can see I updated this test to attempt to use an `await` but none of my changes allow it to work.

**Pull request merged**